code/obj/ToSplit/barber_shop.dm

```diff
@@ -123,200 +185,137 @@
-		M.bioHolder.mobAppearance.customization_first_color = src.customization_first_color
-		M.bioHolder.mobAppearance.customization_second_color = src.customization_first_color
-		M.set_face_icon_dirty()
-		M.set_body_icon_dirty()
-		M.update_clothing()
-		src.empty = 1
-		src.icon_state= "dye-e"
-	//else
-	//	boutput(user, "<span class='alert'>They need to be in a barber chair!</span>")
-
-/////////////////////////////////////////////////////
-//////Scissors Code								/////
-////////////////////////////////////////////////////
-/obj/item/scissors/attack(mob/living/carbon/human/M as mob, mob/user as mob)
-	if (src.remove_bandage(M, user))
-		return
-
-	//If the haircut
-	if (do_haircut(M, user))
-		return
-
-	if (src.reagents && src.reagents.total_volume)
-		logTheThing("combat", user, M, "used [src] on [constructTarget(M,"combat")] (<b>Intent</b>: <i>[user.a_intent]</i>) (<b>Targeting</b>: <i>[user.zone_sel.selecting]</i>) [log_reagents(src)]")
-	else
-		logTheThing("combat", user, M, "used [src] on [constructTarget(M,"combat")] (<b>Intent</b>: <i>[user.a_intent]</i>) (<b>Targeting</b>: <i>[user.zone_sel.selecting]</i>)")
-
-	if (!snip_surgery(M, user))
-		return ..()
-	else
-		if (src.reagents && src.reagents.total_volume)//ugly but this is the sanest way I can see to make the surgical use 'ignore' armor
-			src.reagents.trans_to(M,5)
-		return
-
-
-//should really be moved to progressbar but w/e
-// returns true on success, false on failure
-/obj/item/scissors/proc/do_haircut(mob/living/carbon/human/M as mob, mob/user as mob)
-	if(user.zone_sel.selecting != "head" || user.a_intent != "help")
-		return 0
-
-	if (user == M)
-		boutput(user, "<span class='alert'>You can't cut your own hair!</span>")
-		return 0
-	if(istype(M.buckled, /obj/stool/chair/comfy/barber_chair))
-
-		var/mob/living/carbon/human/H = M
-		if(ishuman(M) && ((H.head && H.head.c_flags & COVERSEYES) || (H.wear_mask && H.wear_mask.c_flags & COVERSEYES) || (H.glasses && H.glasses.c_flags & COVERSEYES)))
-			// you can't stab someone in the eyes wearing a mask!
-			boutput(user, "<span class='notice'>You're going to need to remove that mask/helmet/glasses first.</span>")
-			return 0
-
-		if(M.bioHolder.mobAppearance.customization_first == "None")
-			boutput(user, "<span class='alert'>There is nothing to cut!</span>")
-			return 0
-
-		var/new_style = input(user, "Please select style", "Style")  as null|anything in customization_styles + customization_styles_gimmick
-
-		if (new_style)
-			if(M.bioHolder.mobAppearance.customization_first == "Balding" && new_style != "None")
-				boutput(user, "<span class='alert'>Not enough hair!</span>")
-				return 0
-
-		if(!new_style)
 			return 0
+		var/result_msg1 = "[user] dyes [M]'s hair."
+		var/result_msg2 = "<span class='notice'>You dye [M]'s hair.</span>"
+		var/result_msg3 = "<span class='notice'>[user] dyes your hair.</span>"
+		var/is_barber = user.mind.assigned_role == "Barber" ? 1 : 0
```

Comment:
` ? 1 : 0` is redundant.

Reply:
Just the first conditional returns what I want? Neat!